go/pkg/repository/s3.go

```diff
@@ -295,6 +317,39 @@ func (s *S3Repository) List(dir string) ([]string, error) {
 	return results, err
 }
 
+func (s *S3Repository) ListTarFile(tarPath string) ([]string, error) {
+	// archiver doesn't let us use readers, so download to temporary file
+	// TODO: make a better tar implementation
+	tmpdir, err := files.TempDir("tar")
+	if err != nil {
+		return []string{}, err
+	}
+	defer os.RemoveAll(tmpdir)
+	tmptarball := filepath.Join(tmpdir, filepath.Base(tarPath))
+	if err := s.GetPath(tarPath, tmptarball); err != nil {
+		return []string{}, err
+	}
+	exists, err := files.FileExists(tmptarball)
+	if err != nil {
+		return []string{}, err
+	}
+	if !exists {
+		return []string{}, &DoesNotExistError{msg: "ListTarFile: does not exist: " + tmptarball}
```

Comment:
`nil` instead of `[]string{}`, and without showing the user the internal function name.

Reply:
Fixed.